server/lib/chat/chat-api.ts

```diff
@@ -439,4 +463,182 @@ export class AdminChatApi {
 
     ctx.status = 204
   }
+
+  @httpGet('/banners/:bannerId')
+  @httpBefore(checkAllPermissions('manageChannelContent'))
+  async getChannelBanner(ctx: RouterContext): Promise<AdminGetChannelBannerResponse> {
+    const {
+      params: { bannerId },
+    } = validateRequest(ctx, {
+      params: Joi.object<{ bannerId: ChannelBannerId }>({
+        bannerId: Joi.string().uuid().required(),
+      }),
+    })
+
+    const channelBanner = await adminGetChannelBanner(bannerId)
+
+    if (!channelBanner) {
+      throw new ChatServiceError(
+        ChatServiceErrorCode.ChannelBannerNotFound,
+        'channel banner not found',
+      )
+    }
+
+    const channelIds = new Set<SbChannelId>()
+    for (let j = 0; j < channelBanner.availableIn.length; j++) {
+      channelIds.add(channelBanner.availableIn[j])
+    }
+    const channelInfos = await getChannelInfos(Array.from(channelIds))
```

Comment:
Honestly pretty confused why you need this `Set` stuff at all though, we're pulling this list out of the DB (where it should have already been deduped), so we're deduping a list that we can be pretty sure contains no dupes. Just pass the array you got to this function instead of doing this unnecessary conversion through a Set